### ISSUE TITLE
Materials exploit in the destructive analyzer and experimentor

### ISSUE THREAD

**Issue by Grufus2** [HippieStationArchive/HippieStation13-1#1895]
From this [post
](http://hippiestation.com/viewtopic.php?f=10&t=3517&sid=5a4695632a99ac4c08f9c8e28f871a14)

I think the culprit is this piece of code:

```
rdconsole.dm line 280

if(linked_lathe) //Also sends salvaged materials to a linked protolathe, if any.
                                for(var/material in linked_destroy.loaded_item.materials)
                                    linked_lathe.materials.insert_amount(min((linked_lathe.materials.max_amount - linked_lathe.materials.total_amount), (linked_destroy.loaded_item.materials[material]*(linked_destroy.decon_mod/10))), material)
                                feedback_add_details("item_deconstructed","[linked_destroy.loaded_item.type]")
                            linked_destroy.loaded_item = null
```

especially `linked_destroy.loaded_item.materials[material]*(linked_destroy.decon_mod/10`

As you can see the autolathe receives the amount through the item materials.

A material sheet always have 2000 materials(1 sheet/amount) according to the mineral.

So if you put a material sheet with an amount bigger than zero and less than one. You will have a surplus in that material.

The exploiter can even upgrade the destructo analyzer to get even more materials.

One solution I can see is to put a check if the material amount is bigger than zero and less than one.

edit:

I found the same problem in the experimentor(it is possible if you link the experimentor console with an autolathe)

He also got a NaN (Not a Number), but I don't think he can do much with that(doesn't display that he can eject it).

**Comment by Grufus2** [comment]
I also can see that he found an exploit in the chem dispenser. I don't understand though, the chem dispenser is infinite if you keep waiting.

**Comment by Grufus2** [comment]
If coders want to test it:

Spawn a metal sheet.
Change the amount to like 0.0001
Put in the destructo analyzer.
If the destructo analyzer doesn't have upgrades, it will give you (2000\* 3/10) or 600 material. If it is fully upgraded the decon_mod is 12 which will give you 2000*12/10 or 2400 materials, basically 1.2 sheets of metal
Repeat the cycle to get alot of metal.

**Comment by Carbonhell** [comment]
It should've been fixed in #1939. If the issue arises again make another issue
